src/NDItk/MultigroupTable/src/readOutgoingStructure.hpp

```diff
@@ -0,0 +1,32 @@
+template< typename Iterator >
+void readOutgoingStructure( const std::string& key, Iterator& iter, const Iterator& end ) {
+
+  base::Keyword secondary( key );
+  unsigned int particle = secondary.particle().value();
+  if ( this->metadata_.numberOutgoingGroups( particle ).has_value() ) {
+
+    auto pos = std::lower_bound( this->outgoing_structure_.begin(),
```

Comment:
Why is this part more complicated than with the Primary group structure? Aside from having a particle associated with it, shouldn't it be the same read in G_outgoing + 1 values?

Reply:
I store the primary structure in a record of its own while the outgoing structures get put in a vector. So for the outgoing particles, I need to search for the position where the record needs to be put in. Since the outgoing particle data (excluding the primary particle) are technically optional, separating the primary and outgoing structures makes sense.